src/Libraries/Microsoft.Extensions.DataIngestion.MarkItDown/MarkItDownMcpReader.cs

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using ModelContextProtocol.Client;
+using ModelContextProtocol.Protocol;
+
+namespace Microsoft.Extensions.DataIngestion;
+
+/// <summary>
+/// Reads documents by converting them to Markdown using the <see href="https://github.com/microsoft/markitdown">MarkItDown</see> MCP server.
+/// </summary>
+public class MarkItDownMcpReader : IngestionDocumentReader
+{
+    private readonly Uri _mcpServerUri;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MarkItDownMcpReader"/> class.
+    /// </summary>
+    /// <param name="mcpServerUri">The URI of the MarkItDown MCP server (e.g., http://localhost:3001/sse).</param>
+    public MarkItDownMcpReader(Uri mcpServerUri)
+    {
+        _mcpServerUri = Throw.IfNull(mcpServerUri);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IngestionDocument> ReadAsync(FileInfo source, string identifier, string? mediaType = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        if (!source.Exists)
+        {
+            throw new FileNotFoundException("The specified file does not exist.", source.FullName);
+        }
+
+        // Read file content as base64 data URI
+#if NET
+        byte[] fileBytes = await File.ReadAllBytesAsync(source.FullName, cancellationToken).ConfigureAwait(false);
+#else
+        byte[] fileBytes;
+        using (FileStream fs = new(source.FullName, FileMode.Open, FileAccess.Read, FileShare.Read))
```

Comment:
We need to open the file for async IO (and we can disable the buffering along the way):

```suggestion
        using (FileStream fs = new(source.FullName, FileMode.Open, FileAccess.Read, FileShare.Read, 1, FileOptions.Asynchronous))
```